src/broadcast/listenbrainzlistener/listenbrainzservice.cpp

```diff
@@ -0,0 +1,70 @@
+
+#include <QJsonObject>
+
+#include "preferences/listenbrainzsettings.h"
+#include "broadcast/listenbrainzlistener/listenbrainzservice.h"
+#include "broadcast/listenbrainzlistener/listenbrainzjsonfactory.h"
+#include "broadcast/listenbrainzlistener/networkmanager.h"
+
+ListenBrainzService::ListenBrainzService(UserSettingsPointer pSettings)
+        :  m_request(ListenBrainzAPIURL),
+           m_latestSettings(ListenBrainzSettingsManager::getPersistedSettings(pSettings)),
+           m_COSettingsChanged(kListenBrainzSettingsChanged),
+           m_pCurrentJSON(nullptr) {
+    connect(&m_manager,&QNetworkAccessManager::finished,
+            this,&ListenBrainzService::slotAPICallFinished);
+    connect(&m_COSettingsChanged,&ControlPushButton::valueChanged,
+            this,&ListenBrainzService::slotSettingsChanged);
+    m_request.setHeader(QNetworkRequest::ContentTypeHeader,"application/json");
+    if (m_latestSettings.enabled) {
+        m_request.setRawHeader("Authorization","Token " + m_latestSettings.userToken.toUtf8());
+    }
+}
+
+
+void ListenBrainzService::slotBroadcastCurrentTrack(TrackPointer pTrack) {
+    Q_UNUSED(pTrack);
+    /*if (!pTrack || !m_latestSettings.enabled)
+        return;
+    m_pCurrentJSON = new QByteArray(
+            ListenBrainzJSONFactory::getJSONFromTrack(
+                    pTrack,ListenBrainzJSONFactory::NowListening));
+    m_manager.post(m_request,*m_pCurrentJSON);*/
+}
+
+void ListenBrainzService::slotScrobbleTrack(TrackPointer pTrack) {
+    if (!pTrack || !m_latestSettings.enabled)
+        return;
+    m_pCurrentJSON = new QByteArray(
```

Comment:
There is no need for QByteArray Bering on the heap. You can use = here, and clear() instead of delete below.

Reply:
True, thanks.